File: backend/src/main/java/com/project2/domain/chat/dto/ChatMessageResponseDTO.java
package com.project2.domain.chat.dto;

import com.project2.domain.chat.entity.ChatMessage;
import com.project2.domain.member.dto.MemberDTO;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
public class ChatMessageResponseDTO {
    public final Long id;
    private final MemberDTO sender;
    private final String content;
    private final LocalDateTime createdAt;

    public ChatMessageResponseDTO(ChatMessage chatMessage) {
        this.id = chatMessage.getId();
        this.sender = MemberDTO.Companion.from(chatMessage.getSender());
        this.content = chatMessage.getContent();
        this.createdAt = chatMessage.getCreatedDate();
    }

}

File: backend/src/main/kotlin/com/project2/domain/chat/dto/ChatMessageResponseDTO.kt
package com.project2.domain.chat.dto

import com.project2.domain.chat.entity.ChatMessage
import com.project2.domain.member.dto.MemberDTO
import java.time.LocalDateTime

data class ChatMessageResponseDTO(
        val id: Long,
        val sender: MemberDTO,
        val content: String,
        val createdAt: LocalDateTime
) {
    companion object {
        @JvmStatic
        fun from(chatMessage: ChatMessage): ChatMessageResponseDTO {
            return ChatMessageResponseDTO(
                    id = chatMessage.id!!,
                    sender = MemberDTO.from(chatMessage.sender),
                    content = chatMessage.content,
                    createdAt = chatMessage.createdDate
            )
        }
    }
}